lightkurve/factory.py

```diff
@@ -0,0 +1,441 @@
+        cols.append(fits.Column(name='POS_CORR1', format='E', unit='pixels',
+                                array=self.pos_corr1))
+        cols.append(fits.Column(name='POS_CORR2', format='E', unit='pixels',
+                                array=self.pos_corr2))
+        coldefs = fits.ColDefs(cols)
+        hdu = fits.BinTableHDU.from_columns(coldefs)
+
+        # Set the header with defaults
+        template = self._header_template(1)
+        for kw in template:
+            if kw not in ['XTENSION', 'NAXIS1', 'NAXIS2', 'CHECKSUM', 'BITPIX']:
+                try:
+                    hdu.header[kw] = (self.keywords[kw],
+                                      self.keywords.comments[kw])
+                except KeyError:
+                    hdu.header[kw] = (template[kw],
+                                      template.comments[kw])
+        wcs_keywords = {'CTYPE1': '1CTYP{}',
+                        'CTYPE2': '2CTYP{}',
+                        'CRPIX1': '1CRPX{}',
+                        'CRPIX2': '2CRPX{}',
+                        'CRVAL1': '1CRVL{}',
+                        'CRVAL2': '2CRVL{}',
+                        'CUNIT1': '1CUNI{}',
+                        'CUNIT2': '2CUNI{}',
+                        'CDELT1': '1CDLT{}',
+                        'CDELT2': '2CDLT{}',
+                        'PC1_1': '11PC{}',
+                        'PC1_2': '12PC{}',
+                        'PC2_1': '21PC{}',
+                        'PC2_2': '22PC{}'}
+        # Override defaults using data calculated in from_fits_images
+        for kw in ext_info.keys():
+            if kw in wcs_keywords.keys():
+                for x in [4, 5, 6, 7, 8, 9]:
+                    hdu.header[wcs_keywords[kw].format(x)] = ext_info[kw]
+            else:
+                hdu.header[kw] = ext_info[kw]
+        return hdu
+
+    def _make_aperture_extension(self):
+        """Create the aperture mask extension (i.e. extension #2)."""
+        mask = 3 * np.ones((self.n_rows, self.n_cols), dtype='int32')
+        hdu = fits.ImageHDU(mask)
+
+        # Set the header from the template TPF again
+        template = self._header_template(2)
+        for kw in template:
+            if kw not in ['XTENSION', 'NAXIS1', 'NAXIS2', 'CHECKSUM', 'BITPIX']:
+                try:
+                    hdu.header[kw] = (self.keywords[kw],
+                                      self.keywords.comments[kw])
+                except KeyError:
+                    hdu.header[kw] = (template[kw],
+                                      template.comments[kw])
+
+        # Override the defaults where necessary
+        for keyword in ['CTYPE1', 'CTYPE2', 'CRPIX1', 'CRPIX2', 'CRVAL1', 'CRVAL2', 'CUNIT1',
+                        'CUNIT2', 'CDELT1', 'CDELT2', 'PC1_1', 'PC1_2', 'PC2_1', 'PC2_2']:
+                hdu.header[keyword] = ""  # override wcs keywords
+        hdu.header['EXTNAME'] = 'APERTURE'
+        return hdu
+
+
+    @staticmethod
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [TargetPixelFileFactory.from_fits_images is too complex (19)](https://app.codacy.com/manual/barentsen/lightkurve/pullRequest?prid=4911672)